packages/roosterjs-plugin-picker/lib/PickerPlugin.ts

```diff
@@ -371,34 +411,20 @@ export default class PickerPlugin<T extends PickerDataProvider = PickerDataProvi
                     this.setAriaActiveDescendant(this.dataProvider.getSelectedIndex());
                 }
 
-                this.handleKeyDownEvent(event);
+                this.cancelDefaultKeyDownEvent(event);
             } else if (
                 this.dataProvider.selectOption &&
                 (keyboardEvent.key == ENTER_CHARCODE || keyboardEvent.key == TAB_CHARCODE)
             ) {
                 this.dataProvider.selectOption();
-                this.handleKeyDownEvent(event);
+                this.cancelDefaultKeyDownEvent(event);
             } else {
                 // Currently no op.
             }
         } else {
             if (keyboardEvent.key == BACKSPACE_CHARCODE) {
-                let searcher = cacheGetContentSearcher(event, this.editor);
-                let nodeBeforeCursor = searcher.getInlineElementBefore()
-                    ? searcher.getInlineElementBefore().getContainerNode()
-                    : null;
-                let nodeId = nodeBeforeCursor ? this.getIdValue(nodeBeforeCursor) : null;
-                if (
-                    nodeId &&
-                    nodeId.indexOf(this.pickerOptions.elementIdPrefix) == 0 &&
-                    (searcher.getInlineElementAfter() == null ||
-                        !(searcher.getInlineElementAfter() instanceof PartialInlineElement))
-                ) {
-                    let replacementNode = this.dataProvider.onRemove(nodeBeforeCursor, true);
-                    this.replaceNode(nodeBeforeCursor, replacementNode);
-                    this.editor.select(replacementNode, PositionType.After);
-                    this.handleKeyDownEvent(event);
-                }
+                this.removeNode(event);
+                this.cancelDefaultKeyDownEvent(event);

```

Comment:
In original code, handleKeyDownEvent is only called when "nodeId &&
                    nodeId.indexOf(this.pickerOptions.elementIdPrefix) == 0 &&
                    (searcher.getInlineElementAfter() == null ||
                        !(searcher.getInlineElementAfter() instanceof PartialInlineElement))", but now it is always called. Is that ok?

Reply:
Good catch, I think it should only be called if the node was successfully removed.